oak_remote_attestation_verification/src/rekor.rs

```diff
@@ -178,32 +178,41 @@ impl TryFrom<&LogEntry> for RekorSignatureBundle {
 pub fn verify_rekor_log_entry(
     log_entry_bytes: &[u8],
     pem_encoded_rekor_public_key_bytes: &[u8],
-    pem_encoded_endorser_public_key_bytes: &[u8],
     endorsement_bytes: &[u8],
 ) -> anyhow::Result<()> {
     verify_rekor_signature(log_entry_bytes, pem_encoded_rekor_public_key_bytes)?;
 
-    let parsed: BTreeMap<String, LogEntry> = serde_json::from_slice(log_entry_bytes)
-        .context("couldn't parse bytes into a LogEntry object")?;
-    let entry = parsed.values().next().context("no entry in the map")?;
+    let body = get_rekor_log_entry_body(log_entry_bytes)?;
 
-    // Parse base64-encoded entry.body into an instance of Body.
-    let body_bytes: Vec<u8> = BASE64_STANDARD
-        .decode(entry.body.clone())
-        .context("couldn't decode Base64 signature")?;
-    let body: Body =
-        serde_json::from_slice(&body_bytes).context("couldn't parse bytes into a Body object")?;
+    let pem_encoded_endorser_public_key_bytes = BASE64_STANDARD
+        .decode(body.spec.signature.public_key.content.clone())
+        .context("couldn't decode Base64 endorser public key")?;
 
     // Verify the body in the Rekor LogEntry
     verify_rekor_body(
         &body,
         endorsement_bytes,
-        pem_encoded_endorser_public_key_bytes,
+        &pem_encoded_endorser_public_key_bytes,
     )?;
 
     Ok(())
 }
 
+/// Parses the given bytes into a Rekor `LogEntry` object, and returns its `body` parsed into an
+/// instance of `Body`.
+pub fn get_rekor_log_entry_body(log_entry_bytes: &[u8]) -> anyhow::Result<Body> {
+    let parsed: BTreeMap<String, LogEntry> = serde_json::from_slice(log_entry_bytes)
+        .context("couldn't parse bytes into a LogEntry object")?;
+    let entry = parsed.values().next().context("no entry in the map")?;
+
+    // Parse base64-encoded entry.body into an instance of Body.
+    let body_bytes: Vec<u8> = BASE64_STANDARD
```

Comment:
I believe PEM encoding is already a base64 format. Is another round of base64 encoding needed?

Reply:
Yes, IIUC, the public key in a pem file is base64-encoded, but the file contains other things, like tags, etc, and the entire file can be base64 encoded again. This is the field `body` inside the Rekor log entry, which itself contains a public key field which is a base64 encoding of an x.509 pem file!

To be consistent with this, I chose to include the base64-encoded Rekor public key in the `BinaryAttestation` message.